### PR TITLE
Updates to the Swift generator

### PR DESCRIPTION
**Pull request by thomasameisel**
- Add a `piiKind` property to the enums in `TelemetryValue`
- Remove the `TelemetryValue` initializer. We should specify the type of the value even if the client will eventually convert it to a string
- Add support for specifying the PIIKind in a field's annotation
- Add an option to generate one initializer with all properties, including a `nil` default value for optional properties

### CODE REVIEW

**Comment by benjamin-bader** [comment]
Hey, I don't know if this is supposed to be a private repository, or whether I have access that I shouldn't, but I'm getting notifications about this PR.  If this is indeed supposed to be public, then sorry to bother.

(also, judging from the kinds of changes goin in to the workflow, have you considered using CMake?  It could help by finding and wiring up the dependencies for you, e.g. boost, so that you wouldn't have to mess with cppflags.)

**Comment by thomasameisel** [comment]
@benjamin-bader it looks like you're actually an admin on this repo so I'm assuming that's why you're getting notifications. Sorry about that! I just removed you as an admin as well.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          export CFLAGS="-Wno-deprecated-copy-with-user-provided-copy"
+          export CXXFLAGS="-Wno-deprecated-copy-with-user-provided-copy"
+
+          cd compiler/cpp
+          make
+
+      - name: Add executable permission
+        run: chmod +x ./thrift
+
+      - uses: actions/upload-artifact@v3
```
Comment (GyroJoe) [inline .github/workflows/build.yml]: Does it actually keep the +x bit here? May be a good idea to zip it first.

File: lib/cpp/src/thrift/concurrency/ThreadManager.cpp [hunk 2]
```diff
@@ -110,7 +110,7 @@ class ThreadManager::Impl : public ThreadManager {
     return pendingTaskCountMax_;
   }
 
-  size_t expiredTaskCount() {
+  size_t expiredTaskCount() const {
```
Comment (GyroJoe) [inline lib/cpp/src/thrift/concurrency/ThreadManager.cpp]: Do we need this change still?

File: compiler/cpp/src/thrift/generate/t_swift_generator.cc [hunk 3]
```diff
@@ -947,7 +970,7 @@ void t_swift_generator::telemetry_dictionary_value(ofstream& out, t_type* type,
 
     out << "] = ";
 
-    telemetry_dictionary_value(out, tmap->get_val_type(), "value");
+    telemetry_dictionary_value(out, tmap->get_val_type(), "value", "");
```
Comment (GyroJoe) [inline compiler/cpp/src/thrift/generate/t_swift_generator.cc]: Should this be `nil`?

**Review by GyroJoe — APPROVED** [review]

File: compiler/cpp/src/thrift/generate/t_swift_generator.cc (hunk 3 above)
Comment (thomasameisel) [reply compiler/cpp/src/thrift/generate/t_swift_generator.cc]: I changed to pass in `pii_kind`. I figure if someone wants to set the PIIKind of the telemetry dictionary they should be able to do that

**Pull request merged**